### PR TITLE
LLVM: Implement Command Line intrinsics

### PR DESCRIPTION
**Pull request by Smit-create**
Extends #1108

### CODE REVIEW

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 1]
```diff
@@ -1195,10 +1195,14 @@ LFORTRAN_API void _lpython_set_argv(int32_t argc_1, char *argv_1[]) {
...
+    *res = argc;
 }
 
 LFORTRAN_API char *_lpython_get_argv(int32_t index) {
     return argv[index];
 }
+
+LFORTRAN_API void _lfortran_get_command_argument(int32_t *index, char **res) {
+    *res = argv[*index];
+}
```
Comment (certik) [inline src/libasr/runtime/lfortran_intrinsics.c:1208]: This seems replicating `_lpython_get_argv`, so we should just use it.

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 2]
```diff
@@ -6609,6 +6609,22 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         } else {
             throw CodeGenError("ABI type not implemented yet in SubroutineCall.");
         }
+        std::string sub_name = s->m_name;
+        if (sub_name == "get_command_argument") {
```
Comment (certik) [inline src/libasr/codegen/asr_to_llvm.cpp:6613]: You need to ensure we are dealing with an intrinsic function. I think this should be implemented in the front end runtime library (in C) for now.

**Comment by certik** [comment]
Also I think you need to ensure that `_lpython_set_argv` is being called in `main`.

File: src/libasr/runtime/lfortran_intrinsics.c (hunk 1 above)
Comment (Smit-create) [reply src/libasr/runtime/lfortran_intrinsics.c:1208]: Sure, removed that. Thanks.

**Comment by Smit-create** [comment]
> Also I think you need to ensure that _lpython_set_argv is being called in main.

Yeah, it is called when the `main` function is created in LLVM.

**Review by certik — APPROVED** [review]
I think this looks good.

**Comment by czgdp1807** [comment]
Please do not merge this before the sync PRs go in.

**Review by czgdp1807 — APPROVED** [review]
Rebase on top of latest ``main`` and merge after re-writing the git history.

File: integration_tests/cmd_01.f90 [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 
     count = command_argument_count()
-
-    do i = 0, count
-        call get_command_argument(i, arg)
-        print *, trim(arg)
-    end do
+    print *, "count: ", count
+    call get_command_argument(i, arg)
+    print *, trim(arg)
```
Comment (certik) [inline integration_tests/cmd_01.f90:9]: Why was the loop removed? It doesn't work?

Comment (Smit-create) [reply integration_tests/cmd_01.f90:9]: Yes, that is unrelated to this PR. See #1323

Comment (certik) [reply integration_tests/cmd_01.f90:9]: Ah, ok!

**Pull request merged**